src/classes/issues-processor.ts

```diff
@@ -615,28 +615,63 @@ export class IssuesProcessor {
       )} is: ${LoggerService.cyan(shouldRemoveStaleWhenUpdated)}`
     );
 
-    if (shouldRemoveStaleWhenUpdated) {
-      issueLogger.info(`The stale label should not be removed`);
-    } else {
+    if (!shouldRemoveStaleWhenUpdated) {
+      issueLogger.info(
+        `The stale label should not be removed due to an update`
+      );
+    }
+
+    const shouldRemoveStaleWhenCommented: boolean =
+      this._shouldRemoveStaleWhenCommented(issue);
+
+    issueLogger.info(
+      `The option ${issueLogger.createOptionLink(
+        this._getRemoveStaleWhenCommentedUsedOptionName(issue)
+      )} is: ${LoggerService.cyan(shouldRemoveStaleWhenCommented)}`
+    );
+
+    if (!shouldRemoveStaleWhenCommented) {
       issueLogger.info(
-        `The stale label should be removed if all conditions met`
+        `The stale label should not be removed due to a comment`
       );
     }
 
+    issueLogger.info(`Checking if the stale label should be removed...`);
+    let shouldRemoveStaleLabel = false;
+
     // Should we un-stale this issue?
-    if (shouldRemoveStaleWhenUpdated && issueHasComments) {
+    if (shouldRemoveStaleWhenUpdated && issueHasUpdate) {
+      issueLogger.info(
+        LoggerService.white('├── '),
+        `Remove the stale label since the $$type has an update and the workflow should remove the stale label when updated`
+      );
+      shouldRemoveStaleLabel = true;
+    } else if (shouldRemoveStaleWhenCommented && issueHasComments) {
+      issueLogger.info(
+        LoggerService.white('├── '),
+        `Remove the stale label since the $$type has a comment and the workflow should remove the stale label when commented`
+      );
+      shouldRemoveStaleLabel = true;
+    }
+
+    if (shouldRemoveStaleLabel) {
+      await this._removeStaleLabel(issue, staleLabel, true);
+
       issueLogger.info(
-        `Remove the stale label since the $$type has a comment and the workflow should remove the stale label when updated`
+        LoggerService.white('└── '),
+        `Skipping the process since the $$type is now un-stale`
       );
-      await this._removeStaleLabel(issue, staleLabel);
 
       // Are there labels to remove or add when an issue is no longer stale?
       await this._removeLabelsWhenUnstale(issue, labelsToRemoveWhenUnstale);
       await this._addLabelsWhenUnstale(issue, labelsToAddWhenUnstale);
```

Comment:
There is the error which triggered the failing test. The option `labelsToAddWhenUnstale` is empty and therefore no label will be added.